src/iuse.cpp

```diff
@@ -3809,6 +3809,60 @@ int iuse::mp3_on(player *p, item *it, bool t, const tripoint &pos)
     return it->type->charges_to_use();
 }
 
+int iuse::gasmask(player *p, item *it, bool, const tripoint& )
+{
+    std::string oname = it->typeId() + "_on";
+    p->add_msg_if_player(_("Your %1$s is ready to use."), it->tname().c_str());
+    it->convert( oname ).active = true;
+    return it->type->charges_to_use();
+}
+
+int iuse::gasmask_on(player *p, item *it, bool t, const tripoint &pos)
+{
+    static int gas_absorbed; // amount of absorbed gas for current filter charge
```

Comment:
This is not initialized all reads from it (including things like `+=`) yield undefined behaviour.

`static` does not work the way you think it works. The variable is shared by *all* invocations of this function, may it be a NPC, another character and by all items its invoked with. It's also not save anywhere, so it will be "reset" (if it had an initialization) upon every game start. This needs to be a property of the item or similar.

Reply:
I am going to rewrite the entire function so it will be resolved.